File: RedSheet.DbEntities/Models/vNegotiationType.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vNegotiationTypes", Schema = "dbo")]
	[Serializable]
	public partial class vNegotiationType  
	{
 
	[Required] 
	[MaxLength(200)]
	public string ClassName { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int NegotiationTypeId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string NegotiationTypeName { get; set; } 

	public vNegotiationType()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-negotiation-type.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vNegotiationType {
    constructor(vNegotiationType?: vNegotiationType )  {
        let properties = [ "className", "negotiationTypeId", "negotiationTypeName",];
        for (let property of properties)
            if (vNegotiationType && vNegotiationType[property])
                this[property] = vNegotiationType[property];
    }
 
    @required()
    @maxLength(200)
	className : string =   undefined;
 
	negotiationTypeId : number =   0 ;
 
    @required()
    @maxLength(200)
	negotiationTypeName : string =   undefined;


}
